tools/db_bench_tool.cc

```diff
@@ -4217,12 +4217,22 @@ class Benchmark {
         table_options->filter_policy = BlockBasedTableOptions().filter_policy;
       } else if (FLAGS_bloom_bits == 0) {
         table_options->filter_policy.reset();
+      } else if (FLAGS_use_block_based_filter) {
+        // Use back-door way of enabling obsolete block-based Bloom
```

Comment:
Should this also print a warning about using a deprecated flag/option?

Reply:
I don't think we need to be so cautious with RocksDB developers. The connection to public API is loose, and digging into the connection would reveal the current status.

Reply:
How would we go about deprecating command line arguments?  Or do we expect them to "live forever" and just not do anything?

Reply:
For a developer tool like this, I recommend simply removing options that do nothing. For now, the command line option still works, e.g. for the performance test in this PR. In some months, we might be able to remove the "back door" option and the db_bench command line option. In some more months, we can arguably remove read support for block-based filter, as it's not critical data, just a performance booster.